src/cmd-tag

```diff
@@ -0,0 +1,211 @@
+
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    if not avail_tags:
+        print("No tags found")
+        sys.exit(0)
+
+    for t in build_data["tags"]:
+        print(f"name: {t['name']}\n"
+              f"created: {t['created']}\n"
+              f"target: {t['target']}\n")
+    sys.exit(0)
+
+
+def cmd_create(args):
+    """Create a new tag from a build ID"""
+
+    # To create a tag, just append to existing list
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    avail_builds = available_builds(build_data)
+
+    if args.tag in avail_tags:
+        fatal("Cannot create tag that already exists")
+
+    if not args.force and args.build not in avail_builds:
+        fatal("Cannot create a tag with a build that does not exist")
+
+    if args.force:
+        print("INFO: Forcing creation of tag with non-existent "
+              f"build ID: {args.build}")
+
+    if "tags" not in build_data.keys():
+        build_data["tags"] = []
+
+    # Create new tag with timestamp
+    created = rfc3339_time()
+    build_data['tags'].append({"name": args.tag,
+                               "created": created,
+                               "target": args.build})
+
+    # Write out JSON
+    finalize_json(build_data)
+
+
+def cmd_update(args):
+    """Update a tag to new build ID"""
+
+    # To update a tag, iterate through existing tags list, replacing
+    # the one entry that we want
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+    avail_builds = available_builds(build_data)
+
+    if args.tag not in avail_tags:
+        fatal("Cannot update a tag that does not exist")
+
+    if not args.force and args.build not in avail_builds:
+        fatal("Cannot update a tag with a build that does not exist")
+
+    if args.force:
+        print("INFO: Updating tag with non-existent build ID: "
+              f"{args.build}")
+
+    # Create a temp list of tags to generate new set of tags
+    tmptags = []
+    created = rfc3339_time()
+    for t in build_data['tags']:
+        if t['name'] == args.tag:
+            tmptags.append({'name': args.tag,
+                            'created': created,
+                            'target': args.build})
+        else:
+            tmptags.append(t)
+
+    build_data['tags'] = tmptags
+
+    # Write out JSON
+    finalize_json(build_data)
+
+
+def cmd_delete(args):
+    """Delete a tag from build metadata"""
+
+    # To delete a tag, iterate through existing tags list, and
+    # drop the entry we want
+    build_data = init_build_data()
+    avail_tags = available_tags(build_data)
+
+    if args.tag not in avail_tags:
+        fatal("Cannot delete a tag that does not exist")
+
+    # Create a temp list of tags to generate new set of tags
+    tmptags = []
+    for t in build_data['tags']:
+        if t['name'] != args.tag:
+            tmptags.append(t)
+
+    build_data['tags'] = tmptags
```

Comment:
Minor: a list comprehension would work well here, e.g. `build_data['tags'] = [t for t in build_data['tags'] if t['name'] != args.tag]`

Reply:
Ah that is more Pythonic, right?  I'm terrible with list comprehension, but it would clean up that portion.